src/data/list/basic.lean

```diff
@@ -2540,7 +2540,7 @@ end
 @[to_additive]
 lemma head_mul_tail_prod' [monoid α] (L : list α) :
   (L.nth 0).get_or_else 1 * L.tail.prod = L.prod :=
-by { cases L, { simp, refl, }, { simp, }, }
+by cases L; simp
 
 lemma head_add_tail_sum (L : list ℕ) : L.head + L.tail.sum = L.sum :=
 by { cases L, { simp, refl, }, { simp, }, }
```

Comment:
You could probably shorten this one as well.

Reply:
I guess you mean the proof of `head_add_tail_sum`? It apparently relies on `default nat = 0`, which `simp` doesn't close. Anyway I think this is beyond the scope of this PR (I only touched this file because the non-terminal `simp` caused an error with `option.get_or_else_none` added to the simp-set.)